pkg/cmd/infra/router/template.go

```diff
@@ -316,7 +321,58 @@ func (o *TemplateRouterOptions) Run() error {
 		if useProxy := util.Env("ROUTER_USE_PROXY_PROTOCOL", ""); useProxy == "true" || useProxy == "TRUE" {
 			check = metrics.ProxyProtocolHTTPBackendAvailable(u)
 		}
-		metrics.Listen(o.ListenAddr, o.StatsUsername, o.StatsPassword, check)
+
+		kubeconfig := o.Config.KubeConfig()
+		client, err := authorizationclient.NewForConfig(kubeconfig)
+		if err != nil {
+			return err
+		}
+		authz, err := authorizerfactory.DelegatingAuthorizerConfig{
+			SubjectAccessReviewClient: client.SubjectAccessReviews(),
+			AllowCacheTTL:             2 * time.Minute,
+			DenyCacheTTL:              5 * time.Second,
+		}.New()
+		if err != nil {
+			return err
+		}
+		tokenClient, err := authenticationclient.NewForConfig(kubeconfig)
+		if err != nil {
+			return err
+		}
+		authn, _, err := authenticatorfactory.DelegatingAuthenticatorConfig{
+			Anonymous:               true,
+			TokenAccessReviewClient: tokenClient.TokenReviews(),
+			CacheTTL:                10 * time.Second,
+			ClientCAFile:            util.Env("ROUTER_METRICS_AUTHENTICATOR_CA_FILE", ""),
+		}.New()
+		if err != nil {
+			return err
+		}
+		l := metrics.Listener{
+			Addr:          o.ListenAddr,
+			Username:      o.StatsUsername,
+			Password:      o.StatsPassword,
+			Authenticator: authn,
+			Authorizer:    authz,
+			Record: authorizer.AttributesRecord{
+				ResourceRequest: true,
+				APIGroup:        "route.openshift.io",
+				Resource:        "routers",
+				Name:            o.RouterName,
+			},
+			Checks: []healthz.HealthzChecker{check},
+		}
+		if certFile := util.Env("ROUTER_METRICS_TLS_CERT_FILE", ""); len(certFile) > 0 {
```

Comment:
Why is this an env var and not a mounted file?

Reply:
> Why is this an env var and not a mounted file?

Oh, nm. Misread.  It's the name.  Why isn't that a flag?

Reply:
Following convention mostly for things we aren't sure we want a formal API in flags for.  I could set it experimental as a flag I guess.